src/fetch.js

```diff
@@ -247,5 +262,24 @@ const getSyncData = async (
       aggregatedResponse
     );
   }
+
+  if (response.sync_token) {
+    /**
+     * To make final sync call and concatenate the result if found any during on fetch request.
+     */
+    const aggregatedSyncToken = syncToken.filter(item => item !== undefined);
+    for (const token of aggregatedSyncToken) {
+      const syncResponse = await fetchCsData(
```

Comment:
can you use Promise.all() in case the  `fetchCsData` is not dependent on previous response. that will result in faster completion